### PR TITLE
Implement transaction builder

### PR DESCRIPTION
**Pull request by connor4312**
This lets us do things such as distributed locks, like so:

```js
function lock() {
  return client.if('my_lock', 'createdAt', '==', 0)
    .then(client.put('my_lock').value(id))
    .else(client.get('my_lock'))
    .commit()
    .then(result => console.log(result.succeeded === id ? 'lock acquired' : 'already locked'));
}

function unlock() {
  return client.if('my_lock', 'value', '==', 0)
    .then(client.delete().key('my_lock'))
    .commit();
}
```

### CODE REVIEW

File: src/lease.ts [hunk 1]
```diff
@@ -48,6 +48,32 @@ const enum State {
 
 /**
  * Lease is a high-level manager for etcd leases.
+ * Leases are great for things like service discovery:
```
Comment (SimonSchick) [inline src/lease.ts]: rm?

Comment (SimonSchick) [inline src/lease.ts]: Or make that `@example`

Comment (connor4312) [inline src/lease.ts]: yea....

Comment (connor4312) [inline src/lease.ts]: Typedoc in general is disappointing, but there's no way I'm going to manually typedef all those RPC interfaces.

**Pull request merged**